### PR TITLE
add tests on BitwiseOp::Shl

### PR DESCRIPTION
**Pull request by 6293**

### CODE REVIEW

File: tests/asm.rs [hunk 1]
```diff
@@ -546,3 +546,72 @@ fn bytes_rev() {
...
+    }
+    aluasm_succ! {
+        put    r8192[0],1;
+        put    a16[1],2;
+        put    r8192[2],4;
+        shl    a16[1],r8192[0];
+        eq.n   r8192[0],r8192[2];
+        ret;
+    }
+    // FIXME shift value is limited to 1024
```
Comment (6293) [inline tests/asm.rs]: https://github.com/AluVM/rust-aluvm/issues/88

File: src/pipelines/compiler.rs [hunk 2]
```diff
@@ -505,6 +505,7 @@ impl<'i> Statement<'i> {
     }
 
     fn goto(&'i self, no: u8, issues: &mut Issues<'i, issues::Compile>) -> Option<String> {
+        println!("aaaaaaaaaaaa {:?}", self.operands.get(0));
```
Comment (dr-orlovsky) [inline src/pipelines/compiler.rs]: I think we need to remove this

File: src/pipelines/compiler.rs [hunk 3]
```diff
@@ -529,7 +530,7 @@ impl<'i> Statement<'i> {
                     SemanticError::OperandMissed {
                         operator: self.operator.0,
                         pos: no + 1,
-                        expected: "goto statement",
+                        expected: "goto statemelllnt",
```
Comment (dr-orlovsky) [inline src/pipelines/compiler.rs]: Looks like a typo / accidental change

Comment (6293) [reply src/pipelines/compiler.rs]: thank you

**Comment by dr-orlovsky** [comment]
Why do the tests are failing?

**Comment by 6293** [comment]
dependent on https://github.com/AluVM/rust-aluvm/pull/91

**Comment by dr-orlovsky** [comment]
With https://github.com/AluVM/rust-aluvm/pull/91 merged we can rebase now and then I will merge this

**Comment by 6293** [comment]
rebased

**Review by dr-orlovsky — APPROVED** [review]
ACK https://github.com/AluVM/aluasm/commit/51fcc8af9a813bfee44557c18b8a3c486b5bbf31

**Pull request merged**